python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -271,20 +344,38 @@ def runQuantum(self, task, quantum, taskDef, butler):
         # Get the input and output references for the task
         inputRefs, outputRefs = taskDef.connections.buildDatasetRefs(quantum)
 
-        # Call task runQuantum() method. Any exception thrown by the task
-        # propagates to caller.
-        task.runQuantum(butlerQC, inputRefs, outputRefs)
+        # Call task runQuantum() method.  Catch a few known failure modes and
+        # translate them into specific
+        try:
+            task.runQuantum(butlerQC, inputRefs, outputRefs)
+        except NoWorkFound as err:
+            # Not an error, just an early exit.
+            _LOG.info("Task '%s' on quantum %s exited early: %s",
+                      taskDef.label, quantum.dataId, str(err))
+            pass
+        except RepeatableQuantumError as err:
+            if self.exitOnKnownError:
+                _LOG.warning("Caught repeatable quantum error for %s (%s):", taskDef, quantum.dataId)
+                _LOG.warning(err, exc_info=True)
+                sys.exit(err.EXIT_CODE)
+            else:
+                raise
```

Comment:
Im not sure what the difference is here, this will bring it down in their case right? Is it that EXIT_CODE is something other than say 1?

Reply:
Yes; I'm trying to give BPS a chance to do special things for special exceptions, and that means translating exceptions to exit codes (it's also a convenient way to tell the multiprocessing executor what exception type brought down a worker node, though I'm sure there are other ways to do that).